src/Sema.zig

```diff
@@ -12637,6 +12651,174 @@ fn zirReify(sema: *Sema, block: *Block, inst: Zir.Inst.Index) CompileError!Air.I
     }
 }
 
+fn reifyTuple(
+    sema: *Sema,
+    block: *Block,
+    src: LazySrcLoc,
+    fields_val: Value,
+) CompileError!Air.Inst.Ref {
+    const fields_len = try sema.usizeCast(block, src, fields_val.sliceLen());
+    if (fields_len == 0) return Air.Inst.Ref.empty_struct;
```

Comment:
The only problem with this is that it returns `false` for isTuple:

```zig
    pub fn isTuple(ty: Type) bool {
        return switch (ty.tag()) {
            .tuple, .empty_struct_literal => true,
            else => false,
        };
    }
```

Two options to fix this:

1. Add a new Air.Inst.Ref tag for `empty_struct_literal`
2. Do the `addConstant` dance.

----

After typing this up I realized a mistake I made in my previous comment. `Air.Inst.Ref.empty_struct` is a *value*, not a *type*. So that won't work. Both cases will have to use `addConstant(Type.type, ...)`, with `Value.Tag.ty`. (either that or adding more Value Tags such as `empty_struct_type` and `empty_tuple_type`.

Reply:
Ah yes. Thanks. I’ll fix this up tomorrow.

Reply:
Okay I got this fixed. 

For tuples, I just didn't specially handle `fields_len == 0` because tuples don't require a decl anyways so the logic below that is more or less free and not worth the special handling.

For structs, I created a `empty_struct_literal` type.